//ProtocolBuffers Version
syntax = "proto3";

//This option specifies whats package name of the automatically generated go files
option go_package = "protos/addressServiceProto";

//Definition of the service
service AddressService {
    rpc LookupAddress(Person) returns (Address) {}
    rpc GetAllAddresses(Empty) returns (stream Address) {}
    rpc GetAddressList(Empty) returns (AddressList) {}
}

//Data structures of the service
message Person {
    string familyName = 1;
    string givenName = 2;
}

message Address {
    Person resident = 1;
    string street = 2;
    int32 house = 3;
    int32 zipCode = 4;
    string city = 5;
}

message AddressList {
    repeated Address addrList = 1;
}

message Empty {}